### ISSUE TITLE
[Proposal] Consolidate Main, Config, HWE, Akmods, and Kernel Cache

### ISSUE THREAD

**Issue by m2Giles** [ublue-os/main#691]
As Aurora/Bluefin/Bazzite/Ucore developed, we have dropped usage of hwe internally. Nvidia/asus/surface/hwe images are generated in downstream images using the scripts directly instead of using HWE as a caching layer. Asus and surface images have not been building for F41.

Proposal: Drop HWE repo and just build Nvidia images in main. Do not build images for Asus/Surface.

Config is currently it's own repo. It effectively generates 6 RPMs for consumption and one of them should actually be a file copy. Building RPMs is the correct way to include additional items for our images; however, we have sprawled quite a bit on what we do inside of config.

Proposal: Drop Config repo and include in main.

Akmods needs to be tied to kernel-cache. We have already agreed to merged them but have yet to implement.

Proposal: Continue with Merge and consolidate into main.

This will mean that effectively all of ublue that we and others consume would live in this one Repo.

Negatives: Issues board will be have more traffic. This could be resolved with tagging.

**Comment by bsherman** [comment]
Updated thoughts on this proposal.

Current state of things:
- `hwe` has been announced deprecated as of F42. We struggled to maintain it, so it's going away. `nvidia` image builds (and the `nvidia-install.sh` script) will get moved to `main`
- `kernel-cache` and `akmods` have already been merged into combined `akmods` repo and images (see https://github.com/ublue-os/akmods/pull/295)
- no significant changes have been made to `config` related to this proposal


My suggestions are:
1. Take the next step to cleanup `config` and `main` by merging them. 
2. The `config` RPM artifacts should build on the same fedora release as the main image (currently they are all building on older which seems off)
3. After merging `config` and `main`, move `*-nvidia` image builds and `nvidia-install.sh` to `main` repo.
4. Archive HWE with F42 release as currently planned.
5. Hold off on any further `akmods` / `main` merge consideration until this set of changes has settled out.

**Comment by castrojo** [comment]
Updates to this, ublue-os/config has been merged into ublue-os/packages. This is to centralize it since we're generating RPMs out of all the rules.

**Comment by m2Giles** [comment]
Yes, we ended up going a slightly different direction with Config than a straight merge into main; however, it is still consolidation and centralization.

**Comment by brontium** [comment]
Have there been any further developments on this? Specifically re `-nvidia` builds, it seems `latest` has moved to F42 but `gts` has remained on F40. I  appreciated that one could use the `41` label but the `gts` label has the obvious benefits.

**Comment by bsherman** [comment]
> Have there been any further developments on this? Specifically re `-nvidia` builds, it seems `latest` has moved to F42 but `gts` has remained on F40. I  appreciated that one could use the `41` label but the `gts` label has the obvious benefits.

Still in the works, but I appreciated the reminder yesterday. 

I knew we still needed to finish this issue but didn't realize that HWE repo had no updates for F42. 

The reason for this is it's been quite a while since any of the ublue downstreams (Aurora, Bazzite, Bluefin) have depended on the HWE repo's images. 

All said, I've picked up this task again. This issue will close when we have working silverblue-nvidia images building out of the main repo.

**Comment by bsherman** [comment]
I believe all our stated goals have been completed.
